### DISCUSSION TITLE
Get fields from an array of objects of a json input

### DISCUSSION THREAD

**Discussion by ifraixedes** [TomWright/dasel#80] [Q&A] [answered]
How to get fields from an a `json` array?

I have this input

```json
{
  "items": [
    { "kind": "Pod" },
    { "kind": "Image" }
  ]
}
```
with `jq` I do ` jq <o.json  ".items[].kind"`

but I cannot do that with `dasel` with that syntax, how can I do it?

Many thanks in advance.

**Comment by TomWright** [comment]
@ifraixedes I have converted this to a Q&A discussion as it is not a feature request.

**Reply by ifraixedes** [reply]
Oh sorry, I didn't realize that GitHub has now that feature, I was creating the issue and I realized that I could only create 2 types of issues and any of them were for asking a question, but I understand now why.
My apologies for the inconvenience.

**Comment by TomWright** [comment, marked as answer]
To do this you should use the [any index selector](https://github.com/TomWright/dasel#any-index) in combination with the [multi flag](https://github.com/TomWright/dasel#-m---multiple).

The full selector looks like:
```
.items.[*].kind
```

Here's an example on the playground: https://dasel.tomwright.me/s/d6f56db3-bc64-4f3f-9529-f617e6483dcc

The full command will look like this:
```
dasel -f o.json -m '.items.[*].kind'
```

If you don't want the resulting lines to be quoted you can use the [`--plain` flag](https://github.com/TomWright/dasel#--plain).

**Reply by TomWright** [reply]
@ifraixedes Please let me know if this does/doesn't solve your issue.

**Reply by ifraixedes** [reply]
Unless that I'm doing something wrong or missing something, I'm getting an error

```
➜  cat o.json                                                                                                                                                      git:(master) ✗  
{
  "items": [
    { "kind": "Pod" },
    { "kind": "Image" }
  ]
}
➜  dasel -f o.json '.items.[*].kind'                                                                                                                               git:(master) ✗  
Error: could not query node: could not find value: selector is not supported here: .[*].kind
Usage:
  dasel select -f <file> -p <json,yaml> -s <selector> [flags]

Flags:
  -f, --file string       The file to query.
  -h, --help              help for select
  -m, --multiple          Select multiple results.
  -p, --parser string     Shorthand for -r FORMAT -w FORMAT.
      --plain             Alias of -w plain
  -r, --read string       The parser to use when reading.
  -s, --selector string   The selector to use when querying the data structure.
  -w, --write string      The parser to use when writing.

Error: could not query node: could not find value: selector is not supported here: .[*].kind
```

**Reply by TomWright** [reply]
Sorry I missed the `-m` flag from my example. Please try again

**Reply by ifraixedes** [reply]
Brilliant, that was it!

Many thanks for your help and sorry for not paying good attention to the command line flags.

**Reply by tallgrrl** [reply]
Tom, I have a very similar case, and this is the closest answer I can find. I need the above scenario, but I need to pull back multiple values per line:

{
	"body": [{
			"id": 1,
			"color": "red",
			"value": "#f00"
		},
		{
			"id": 2,
			"color": "green",
			"value": "#0f0"
		},
		{
			"id": 3,
			"color": "blue",
			"value": "#00f"

		}
	]
}

I am using (as per your example)

echo $json | dasel -r json  -m '.body.[*]' -w csv

and the result is 

map[color:red id:1 value:#f00] 
map[color:green id:2 value:#0f0] 
map[color:blue id:3 value:#00f]


(note no commas in the csv)
not 

color,id,value
red,1,#f00
green,2,#0f0
blue,3,#00f

 as I am hoping for. what am I missing?
